components/dashboard/src/start/VerifyModal.tsx

```diff
@@ -0,0 +1,221 @@
+        const isTokenFilled = () => {
+            return state.token && /\d{6}/.test(state.token);
+        };
+        const verifyToken = async () => {
+            const verified = await getGitpodService().server.verifyPhoneNumberVerificationToken(
+                state.phoneNumber!,
+                state.token!,
+            );
+            if (verified) {
+                setState({
+                    ...state,
+                    verified: true,
+                    message: undefined,
+                });
+            } else {
+                setState({
+                    ...state,
+                    message: {
+                        type: "error",
+                        text: `Invalid verification code.`,
+                    },
+                });
+            }
+            return verified;
+        };
+
+        const reset = () => {
+            setState({
+                ...state,
+                sent: undefined,
+                message: undefined,
+                token: undefined,
+            });
+        };
+        return (
+            <Modal
+                onClose={() => {}}
+                closeable={false}
+                onEnter={verifyToken}
+                title="User Validation Required"
+                buttons={
+                    <div>
+                        <button className="ml-2" disabled={!isTokenFilled()} onClick={verifyToken}>
+                            Validate Account
+                        </button>
+                    </div>
+                }
+                visible={true}
+            >
+                <Alert type="warning" className="mt-2">
+                    To use Gitpod you'll need to validate your account with your phone number. This is required to
+                    discourage and reduce abuse on Gitpod infrastructure.
+                </Alert>
+                <div className="pt-4">
+                    <button className="gp-link" onClick={reset}>
+                        &larr; Use a different phone number
+                    </button>
+                </div>
+                <div className="pt-4">
+                    Enter the verification code we sent to {state.phoneNumber}.<br />
+                    Having trouble?{" "}
+                    <a className="gp-link" href="https://www.gitpod.io/contact/support">
+                        Contact support
+                    </a>
+                </div>
+                {state.message ? (
+                    <Alert type={state.message.type} className="mt-4 py-3">
+                        {state.message.text}
+                    </Alert>
+                ) : (
+                    <></>
+                )}
+                <div className="mt-4">
+                    <h4>Verification Code</h4>
+                    <input
+                        autoFocus={true}
+                        className="w-full"
+                        type="text"
+                        placeholder="enter token send via sms"
```

Comment:
Regarding https://github.com/gitpod-io/gitpod/pull/12258#discussion_r952581084:

> We only support the current version of chrome, firefox and safari.

Agree, makes sense. 🤝 Maybe worth being more specific about this also in [our docs](https://www.gitpod.io/docs/self-hosted/helm-deprecated/requirements#supported-web-browsers). Opened https://github.com/gitpod-io/website/pull/2617 as a follow-up PR but closed in https://github.com/gitpod-io/website/pull/2617#issuecomment-1224271963 as this is already covered by the upcoming compatibility matrix.

❗ **issue**: Still placeholders remain problematic from UX perspective. 

If there's a strong sense in keeping the placeholder, let's at least update the casing and replace the token term, which is a new term for users here.

```suggestion
                        placeholder="Enter code sent via SMS"
```